File: Problems 0000-0099/14. Longest Common Prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # if len(strs) == 0:
        #     return ""

        # shortest = min(strs, key=len)
        # for i, ch in enumerate(shortest):
        #     for s in strs:
        #         if s[i] != shortest[i]:
        #             return shortest[:i]
        # return shortest

        if len(strs) == 0:
            return ""

        strs.sort()
        n = min(len(strs[0]), len(strs[-1]))
        for i in range(n):
            if strs[0][i] != strs[-1][i]:
                return strs[0][:i]
        return strs[0]
            
        

File: Problems 0000-0099/14. Longest Common Prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 1) {
            return strs[0];
        }
        string prefix = "";
        sort(strs.begin(), strs.end());
        int n = strs.size();
        string first = strs[0];
        string last = strs[n-1];
        for (int i = 0; i < min(first.length(), last.length()); i++) {
            if (first[i] != last[i]) {
                return prefix;
            }
            prefix += first[i];
        }
        return prefix;
    }
};